erica/request_processing/erica_input/v2/grundsteuer_input_gebaeude.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional
+
+from pydantic import root_validator
+
+from erica.request_processing.erica_input.v2.possibly_aliased_input import PossiblyAliasedInput
+
+
+class Ab1949(PossiblyAliasedInput):
+    is_ab1949: bool
+
+
+class Baujahr(PossiblyAliasedInput):
+    baujahr: Optional[str]
+
+
+class Kernsaniert(PossiblyAliasedInput):
+    is_kernsaniert: bool
+
+
+class Kernsanierungsjahr(PossiblyAliasedInput):
+    kernsanierungsjahr: Optional[str]
+
+
+class Abbruchverpflichtung(PossiblyAliasedInput):
+    has_abbruchverpflichtung: bool
+
+
+class Abbruchverpflichtungsjahr(PossiblyAliasedInput):
+    abbruchverpflichtungsjahr: Optional[str]
+
+
+class Wohnflaeche(PossiblyAliasedInput):
+    wohnflaeche: int
+
+
+class Wohnflaechen(PossiblyAliasedInput):
+    wohnflaeche1: int
+    wohnflaeche2: int
+
+
+class WeitereWohnraeume(PossiblyAliasedInput):
+    has_weitere_wohnraeume: bool
+
+
+class WeitereWohnraeumeDetails(PossiblyAliasedInput):
+    anzahl: int
+    flaeche: int
+
+
+class Garagen(PossiblyAliasedInput):
+    has_garagen: bool
+
+
+class GaragenAnzahl(PossiblyAliasedInput):
+    anzahl_garagen: int
+
+
+class Gebaeude(PossiblyAliasedInput):
+    ab1949: Ab1949
+    baujahr: Optional[Baujahr]
+    kernsaniert: Kernsaniert
+    kernsanierungsjahr: Optional[Kernsanierungsjahr]
+    abbruchverpflichtung: Abbruchverpflichtung
+    abbruchverpflichtungsjahr: Optional[Abbruchverpflichtungsjahr]
+    wohnflaeche: Optional[Wohnflaeche]
+    wohnflaechen: Optional[Wohnflaechen]
+    weitere_wohnraeume: WeitereWohnraeume
+    weitere_wohnraeume_details: Optional[WeitereWohnraeumeDetails]
+    garagen: Garagen
+    garagen_anzahl: Optional[GaragenAnzahl]
+
+    @root_validator
+    def baujahr_must_be_present_if_ab_1949(cls, values):
+        v = values.get("baujahr")
+        if values.get("ab1949").is_ab1949 is True and (not v or not v.baujahr):
+            raise ValueError('has to be set if is_ab1949')
+        return values
```

Comment:
Why did you decide on a root validator instead of 

```
@validator("baujahr", always=True)
    def baujahr_must_be_present_if_ab_1949(cls, v, values):
        if values.get("ab1949").is_ab1949 is True and (not v or not v.baujahr):
            raise ValueError('has to be set if is_ab1949')
        return v
```
This way I think it's clearer which field is validated?

Reply:
> Why did you decide on a root validator instead of

Because we are validating a field which cannot be validated independently, but  rather depends on another field having been already parsed. If `ab1949` has not already been parsed by the time `baujahr` is being parsed (and validated), we either get false results or an error.

> Also the error message should be easier understandable, because the field is actually included in it. Not sure if this is also the case for a root_validator.

Right, and great catch. Will fix the messages right away.

Reply:
As long as you write the field you are checking after the field you are comparing it to in the declaration, there shouldn't be any problem. I just find the specific validators much easier to read, not only because of the error message.

Reply:
Done.